### ISSUE TITLE
Error configuring queue.minDequeueBatchSize, queue.minDequeueBatchSize.timeout to rsyslogd

### ISSUE THREAD

**Issue by bhatguru** [rsyslog/rsyslog#5101]
### Expected behavior

Below flags in action queues and ruleset queues should work as per the doc
https://www.rsyslog.com/doc/v8-stable/rainerscript/queue_parameters.html
`queue.minDequeueBatchSize
queue.minDequeueBatchSize.timeout`

### Actual behavior
Getting error post configuring the above flags to /etc/rsyslog.d/logs.conf
ERROR:-

error during parsing file /etc/rsyslog.d/logs.conf, on or before line 40: parameter 'queue.mindequeuebatchsize.timeout' not known -- typo in config file?

error during parsing file /etc/rsyslog.d/logs.conf, on or before line 40: parameter 'queue.mindequeuebatchsize' not known -- typo in config file?

### Environment
- rsyslog version: rsyslogd 8.32.0
- platform: Ubuntu 18.04.6 LTS
- for configuration questions/issues, include rsyslog.conf and included config files

attached configs works fine without queue.minDequeueBatchSize, queue.minDequeueBatchSize.timeout flags.

**Comment by rgerhards** [comment]
This looks like there is a config error. I don't see an attached config.

**Comment by bhatguru** [comment]
Oh, so sorry something wrong, let me share it again.

PFB:-
[logs.conf.txt](https://github.com/rsyslog/rsyslog/files/11016609/logs.conf.txt)
[rsyslog.conf.txt](https://github.com/rsyslog/rsyslog/files/11016610/rsyslog.conf.txt)

**Comment by rgerhards** [comment]
oh, sry, I overlooked the rsyslog version. It's veeeeery old. I guess the error message is plain right: config param is not known to that version.

**Comment by rgerhards** [comment]
yes, I checked: 8.32 does not support it,neither 8.40. In 8.2022.0 it is present. I didn't dig when it was actually added.

I suggest you use the docs from the distro 8.32 package - it will list the parameters actually supported by old versions. The web is always for rsyslog current.

**Comment by bhatguru** [comment]
Upgraded to **8.2302.0 (aka 2023.02)**, parameters supported.

**Comment by davidelang** [comment]
just a note that having too many queues can slow things down (locking the queues 
to move messages from one queue to the next), you probably don't want both a 
ruleset queue and an action queue. having the ruleset queue for an input is a 
good idea, adding an action queue would only be needed if you have multiple 
actions in the ruleset that you need to decouple.

David Lang

**Comment by bhatguru** [comment]
Yeah, sure thank you @davidelang 

@rgerhards 
Post upgrading to latest version which is 8.2302.0 (aka 2023.02), getting following errors, as per the documentation parameters that i have added in the config looks fine. 

rsyslogd[852]: error during parsing file /etc/rsyslog.d/logs.conf, on or before line 39: parameter 'queue.maxFileSize' not known -- typo in config file? [v8.2302.0 try https://www.r
rsyslogd[852]: error during parsing file /etc/rsyslog.d/logs.conf, on or before line 39: parameter 'queue.maxDiskSpace' not known -- typo in config file? [v8.2302.0 try https://www.
rsyslogd[852]: error during parsing file /etc/rsyslog.d/logs.conf, on or before line 39: parameter 'queue.type' not known -- typo in config file? [v8.2302.0 try https://www.rsyslog.
 rsyslogd[852]: error during parsing file /etc/rsyslog.d/logs.conf, on or before line 39: parameter 'queue.filename' not known -- typo in config file? [v8.2302.0 try https://www.rsys
rsyslogd[852]: imudp: deprecated parameter inputname used. Suggest to use name instead [v8.2302.0 try https://www.rsyslog.com/e/2307 ]

**Comment by davidelang** [comment]
check for missing or extra quotes/prarens/braces before that line.

David Lang

**Comment by rgerhards** [comment]
This looks a bit like the params are at the wrong place. @bhatguru pls share your new config.

**Comment by bhatguru** [comment]
@rgerhards 
Config is exactly same as the one which is attached earlier (logs.conf), but some how after couple of restarts not seeing that error now, haven't changed any parameters for now.

**Comment by bhatguru** [comment]
@rgerhards any reference for sample configuration to ship logs from  (imfile) to (omfwd) over tcp, with latest version, existing configuration is some how not working, no tcp packets being sent from the rsyslogd.

**Comment by rgerhards** [comment]
some samples in the doc, some in the testbench. But if you really didn't change the config, your's should work as well. 
Please create a debug log. Instructions: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html

Be sure that the problem in question occurs at least once during the debug run. Once this is done, please post the debug log (e.g. via pastbin or github). After that, we can review the contents and check how exactly rsyslog processing worked.

**Comment by bhatguru** [comment]
Please find attached debug logs @rgerhards 


[rsyslog.debug.txt](https://github.com/rsyslog/rsyslog/files/11058834/rsyslog.debug.txt)

**Comment by rgerhards** [comment]
The debug log does not contain these error messages. Did you put the debug enable statements at the top of rsyslog.conf?

**Comment by rgerhards** [comment]
oh, and rsyslog is an older version:

glbl.c: This is rsyslog version 8.2008.0.9d70869d4ea

**Comment by bhatguru** [comment]
> 

yes

**Comment by bhatguru** [comment]
`The debug log does not contain these error messages. Did you put the debug enable statements at the top of rsyslog.conf?` yes

**Comment by bhatguru** [comment]
```
oh, and rsyslog is an older version:

glbl.c: This is rsyslog version 8.2008.0.9d70869d4ea
```
even same observation with latest version `8.2302.0`

**Comment by rgerhards** [comment]
well, with no errors and from a different machine, the debug log makes no sense. Create one containing the problem, as I requested ;-)

**Comment by ethanxa** [comment]
> yes, I checked: 8.32 does not support it,neither 8.40. In 8.2022.0 it is present. I didn't dig when it was actually added.
> 
> I suggest you use the docs from the distro 8.32 package - it will list the parameters actually supported by old versions. The web is always for rsyslog current.

Hi!I have a similar problem too, the rsyslog version in server is 08.32.0,when I use the following conf(I want to config tls encryption transfer),error noticed that parameter 'StreamDriver.Mode/Name/AuthMode' not known.Where can I find the doc of 08.32.0 about the valid paremeters? Thanks!

the conf:
# Set certificate files
global(
  DefaultNetstreamDriverCAFile="/etc/pki/ca-trust/source/anchors/ca-cert.pem"
  DefaultNetstreamDriverCertFile="/etc/pki/ca-trust/source/anchors/server-cert.pem"
  DefaultNetstreamDriverKeyFile="/etc/pki/ca-trust/source/anchors/server-key.pem"
)

# TCP listener
module(
  load="imtcp"
  PermittedPeer=["client1.example.com", "client2.example.com"]
  StreamDriver.AuthMode="x509/name"
  StreamDriver.Mode="1"
  StreamDriver.Name="ossl"
)

# Start up listener at port 514
input(
  type="imtcp"
  port="514"
)

**Comment by davidelang** [comment]
On Tue, 9 Apr 2024, ethanxa wrote:

>> yes, I checked: 8.32 does not support it,neither 8.40. In 8.2022.0 it is present. I didn't dig when it was actually added.
>>
>> I suggest you use the docs from the distro 8.32 package - it will list the parameters actually supported by old versions. The web is always for rsyslog current.
>
> Hi!I have a similar problem too, the rsyslog version in server is 08.32.0,when 
> I use the following conf(I want to config tls encryption transfer),error 
> noticed that parameter 'StreamDriver.Mode/Name/AuthMode' not known.Where can I 
> find the doc of 08.32.0 about the valid paremeters? Thanks!

there should be a -doc package from your distro where you are getting 8.32.

your other option is that the rsyslog docs are in git, so you could clone the 
repo and go back the 5 or so years to get to the 8.32 docs

David Lang

>
> the conf:
> # Set certificate files
> global(
>  DefaultNetstreamDriverCAFile="/etc/pki/ca-trust/source/anchors/ca-cert.pem"
>  DefaultNetstreamDriverCertFile="/etc/pki/ca-trust/source/anchors/server-cert.pem"
>  DefaultNetstreamDriverKeyFile="/etc/pki/ca-trust/source/anchors/server-key.pem"
> )
>
> # TCP listener
> module(
>  load="imtcp"
>  PermittedPeer=["client1.example.com", "client2.example.com"]
>  StreamDriver.AuthMode="x509/name"
>  StreamDriver.Mode="1"
>  StreamDriver.Name="ossl"
> )
>
> # Start up listener at port 514
> input(
>  type="imtcp"
>  port="514"
> )
>
>

**Comment by ethanxa** [comment]
> On Tue, 9 Apr 2024, ethanxa wrote: > yes, I checked: 8.32 does not support it,neither 8.40. In 8.2022.0 it is present. I didn't dig when it was actually added. > > I suggest you use the docs from the distro 8.32 package - it will list the parameters actually supported by old versions. The web is always for rsyslog current. Hi!I have a similar problem too, the rsyslog version in server is 08.32.0,when I use the following conf(I want to config tls encryption transfer),error noticed that parameter 'StreamDriver.Mode/Name/AuthMode' not known.Where can I find the doc of 08.32.0 about the valid paremeters? Thanks!
> there should be a -doc package from your distro where you are getting 8.32. your other option is that the rsyslog docs are in git, so you could clone the repo and go back the 5 or so years to get to the 8.32 docs David Lang
> […](#)
> the conf: # Set certificate files global( DefaultNetstreamDriverCAFile="/etc/pki/ca-trust/source/anchors/ca-cert.pem" DefaultNetstreamDriverCertFile="/etc/pki/ca-trust/source/anchors/server-cert.pem" DefaultNetstreamDriverKeyFile="/etc/pki/ca-trust/source/anchors/server-key.pem" ) # TCP listener module( load="imtcp" PermittedPeer=["client1.example.com", "client2.example.com"] StreamDriver.AuthMode="x509/name" StreamDriver.Mode="1" StreamDriver.Name="ossl" ) # Start up listener at port 514 input( type="imtcp" port="514" )

thx!I downloaded the rsyslog-8.32.0.tar.gz,and then which path I can find the supported parameters?

**Comment by rgerhards** [comment]
> * should I compile lmnsd_ossl?

no, there are a many other interfaces missing. Just upgrade to the current version.

The rsyslog project provides packages for current versions: https://www.rsyslog.com/downloads/download-other/

**Comment by ethanxa** [comment]
> > * should I compile lmnsd_ossl
> 
> no, there are a many other interfaces missing. Just upgrade to the current version.
> 
> The rsyslog project provides packages for current versions: https://www.rsyslog.com/downloads/download-other/

Thx for advice. now 'could not load module lmnsd_ossl' problem is fixed, then notice 'Certificate file could not be accessed', I am using a existed CA cert file in my linux system on the board.
Do I need to generate CA cert by certtool?

**Comment by ethanxa** [comment]
> > * should I compile lmnsd_ossl?
> 
> no, there are a many other interfaces missing. Just upgrade to the current version.
> 
> The rsyslog project provides packages for current versions: https://www.rsyslog.com/downloads/download-other/

I want to config a Ubuntu18.04 pc as Server to receive TLS log, but Error shows could not load lmnsd_ossl.so.How to enable support openssl in Ubuntu18.04?The rsyslog version is 8.2002.0.

**Comment by rgerhards** [comment]
I tend to think it is not available on that old platform. Can you post at least the exact error message and, better, a startup debug log.

Please create a debug log. Instructions: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html

**Be sure that the problem in question occurs at least once during the debug run**. Once this is done, please post the debug log (e.g. via pastebin or github). After that, we can review the contents and check how exactly rsyslog processing worked.

**Comment by ethanxa** [comment]
> I tend to think it is not available on that old platform. Can you post at least the exact error message and, better, a startup debug log.
> 
> Please create a debug log. Instructions: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html
> 
> **Be sure that the problem in question occurs at least once during the debug run**. Once this is done, please post the debug log (e.g. via pastebin or github). After that, we can review the contents and check how exactly rsyslog processing worked.
[Uploading debug.7z…]()

Here is the conf and the debug file. Thx

**Comment by ethanxa** [comment]
[debug.zip](https://github.com/rsyslog/rsyslog/files/15022818/debug.zip)

**Comment by rgerhards** [comment]
I see that the file itself is actually missing. Did you install the openssl tls driver. It should be in a separate package. If there is none, it's not supported on that platform.
